ocs_ci/ocs/scale_lib.py

```diff
@@ -0,0 +1,385 @@
+                'usage_70_80': '10M', 'usage_80_85': '512K', 'usage_above_85': '10K'}
+            WARN: Make sure dict key is same as above example.
+        Returns:
+            size (str): IO size to be considered for cluster env
+        """
+        osd_dict = cluster.get_osd_utilization()
+        if custom_size_dict:
+            size_dict = custom_size_dict
+        else:
+            size_dict = {
+                'usage_below_40': '1G', 'usage_40_60': '128M', 'usage_60_70': '10M',
+                'usage_70_80': '5M', 'usage_80_85': '512K', 'usage_above_85': '10K'
+            }
+        temp = 0
+        for k, v in osd_dict.items():
+            if temp <= v:
+                temp = v
+        if temp <= 40:
+            size = size_dict['usage_below_40']
+        elif 40 < temp <= 50:
+            size = size_dict['usage_40_50']
+        elif 60 < temp <= 70:
+            size = size_dict['usage_60_70']
+        elif 70 < temp <= 80:
+            size = size_dict['usage_70_80']
+        elif 80 < temp <= 85:
+            size = size_dict['usage_80_85']
+        else:
+            size = size_dict['usage_above_85']
+            logging.warn(f"One of the OSD is near full {temp}% utilized")
+        return size
+
+    def get_rate_based_on_cls_iops(self, custom_iops_dict=None, osd_size=2048):
+        """
+        Function to check ceph cluster iops and suggest rate param for fio.
+        Args:
+            custom_iops_dict (dict): Dictionary of rate param to be used during IO run.
+                eg: iops_dict = {'usage_below_40%': '16k', 'usage_40%_60%': '8k',
+                'usage_60%_80%': '4k', 'usage_80%_95%': '2K'}
+            WARN: Make sure dict key is same as above example.
+            osd_size (int): Size of the OSD in GB
+        Returns:
+            rate_param (str): Rate parm for fio based on ceph cluster IOPs
+        """
+        # Check for IOPs limit percentage of cluster and accordingly suggest fio rate param
+        cls_obj = cluster.CephCluster()
+        iops = cls_obj.get_iops_percentage(osd_size=osd_size)
+        if custom_iops_dict:
+            iops_dict = custom_iops_dict
+        else:
+            iops_dict = {
+                'usage_below_40%': '8k', 'usage_40%_60%': '8k',
+                'usage_60%_80%': '4k', 'usage_80%_95%': '2K'
+            }
+        if (iops * 100) <= 40:
+            rate_param = iops_dict['usage_below_40%']
+        elif 40 < (iops * 100) <= 60:
+            rate_param = iops_dict['usage_40%_60%']
+        elif 60 < (iops * 100) <= 80:
+            rate_param = iops_dict['usage_60%_80%']
+        elif 80 < (iops * 100) <= 95:
+            rate_param = iops_dict['usage_80%_95%']
+        else:
+            logging.warn(f"Cluster iops utilization is more than {iops * 100} percent")
+            raise UnavailableResourceException("Overall Cluster utilization is more than 95%")
+        return rate_param
+
+    def create_scale_pods(
```

Comment:
I think we'll need a variable for the environment as well. For now, it can default to AWS.

Reply:
Yes, for now it's default to AWS and added this check in code.
`        if config.ENV_DATA['deployment_type'] == 'ipi' and config.ENV_DATA['platform'].lower() == 'aws':
            # Create machineset for app worker nodes, which will create one app worker node
            self.ms_name = machine.create_custom_machineset(instance_type=instance_type, zone='a')
            machine.wait_for_new_node_to_be_ready(self.ms_name)
            self.app_worker_nodes = machine.get_machine_from_machineset(self.ms_name)`

Reply:
@keesturam let me know either this check is good enough.

Reply:
A couple of questions:
1) What happens if this test is run on VMware? 
2) what is zone-a? is that availability zone?